### PR TITLE
Initial Django project and API app

### PR DESCRIPTION
**Pull request by willbarton**
Here's an initial Django project and API app. The API app has models for repositories and data points, tests for the models, mock-data sources for generating some random data points to test with, and a basic REST API for both. 

@Ooblioob @mehtadev17, can you review for anything that stands out? It's a lot, but a lot of it is Django boilerplate (pretty much all of porchlight/).

### CODE REVIEW

File: porchlightapi/models.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+from django.db import models
+
+from porchlightapi import settings
+from porchlightapi.utils import get_class_or_func
+
+
+class Repository(models.Model):
+    """
+    A repository of code that is deployed that may or may not have
```
Comment (Ooblioob) [inline porchlightapi/models.py:11]: :+1: for documentation!

**Comment by Ooblioob** [comment]
@willbarton 
I see what you're trying to do here with the `reliable_mock_undeployed_source` method but this may be a place where a formal Mocking framework is more useful.  What we want is a method we can mock in order to force it to return a set value.  We could leave `mock_undeployed_source` (perhaps we rename it) since it generates visually pretty data, but for our tests we should just call the test like this: 

```
@patch("porchlightapi.sources.mock_undeployed_source")
def test_undeployed_value_source(self, mock_undeployed_source):
    mock_undeployed_source.return_value = (value1, value2, value3)
...etc
```

This will give us control over the method even if it's stored in the db (I think, I've never tried it with a method like this).  We can control what it returns and even verify that it was called (when appropriate).  

If we use a mocking class then we don't have to write our own extra stuff and it's a little more standard.  The code you've written is fine for now, but because the "mock" implementation is random it's a nightmare to test and requires us to write multiple "reliable mock" implementations in our unit tests!

Here's the link to the documentation on Mocks  
http://www.voidspace.org.uk/python/mock/getting-started.html#using-mock

**Comment by willbarton** [comment]
@Ooblioob, Sounds good, I'll pull in mock.

File: porchlightapi/tests.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+## Value Data Points
+from porchlightapi.models import ValueDataPoint
+
+class ValueDataPointTestCase(TestCase):
+
+    def setUp(self):
+        create_test_repository()
+
+    def test_value_data_point_creation(self):
```
Comment (Ooblioob) [inline porchlightapi/tests.py]: I see no strong reason we need to make an integration test out of this.  We could accomplish the same by mocking the test_repo object.  Something to the tune of:

```
@patch("porchlightapi.models.Repository")
def test_value_data_point_creation(self, test_repo):
    test_repo.undeployed_value.return_value = some_value
    ...
```

There's also a way to mock `self.create` so we can just call the `assert_called_with()` method to verify it was called with what we wanted.  
http://www.voidspace.org.uk/python/mock/getting-started.html#using-mock

**Comment by willbarton** [comment]
@Ooblioob, when you get a chance, can you review the latest commits? I've switched to using mock (which also makes the test code cleaner, which makes me happy), and I've clarified the `ValueDataPoint` test case by making it a test case for `ValueDataPointManager`, testing that it correctly extracts data from a `Repository` (mocked) and correctly calls the default manager `create()` method (which I'm confirming using mock).

**Comment by Ooblioob** [comment]
@willbarton Very nice job, you get a selfie!  The changes for the mocks look great!  
selfie-3

Prepare to merge!

**Pull request merged**